### DISCUSSION TITLE
Github Discussions for Metalsmith

### DISCUSSION THREAD

**Discussion by webketje** [segmentio/metalsmith#390] [Announcements]
Starting today, Github discussions are enabled for Metalsmith. 
Do you have a question, suggestion, plugin idea, metalsmith build, static-site generator news or other related stuff to share that doesn't quite fit in an issue yet? Share it here!

**Comment by wernerglinka** [comment]
Hi @webketje, good move to GH Discussions. How's the summer in Belgium, hot as hell.? Well, it is in Minnesota. Lots of new Metalsmith blogposts at https://www.glinka.co. Have a look if you have a minute ;-)

**Comment by rludgero** [comment]
### Sugestion

It would be nice to have these on the @metalsmith plugin list.

- Eslint
- Stylelint
- Prettier
- Squish

**Reply by webketje** [reply]
@rludgero stylelint can already be done via [@metalsmith/postcss](https://github.com/metalsmith/postcss/blob/main/test/index.js#L270). I do use prettier & eslint in all repo's, but they usually exceed Metalsmith build scope (repo-wide).Never heard about Squish, and not sure what it is even after searching for it?

**Reply by rludgero** [reply]
> @rludgero stylelint can already be done via [@metalsmith/postcss](https://github.com/metalsmith/postcss/blob/main/test/index.js#L270). I do use prettier & eslint in all repo's, but they usually exceed Metalsmith build scope (repo-wide).Never heard about Squish, and not sure what it is even after searching for it?

Ok, thanks for the clarification about not being able to implement them.

[Squish](https://squish.addy.ie/) is an image compressor which makes use of WebAssembly for high-performance image optimization. 

Here are the [repository](https://github.com/addyosmani/squish)

**Reply by webketje** [reply]
Ah thanks, but that looks like an app rather than a plugin CLI tool. And when I click through (Squish ->jSquash -> Squoosh), it seems the project is unmaintained. I'd been looking at replacing imagemin, but was looking more in the direction of sharp: https://github.com/lovell/sharp which also allows on-the-fly img transforms.

FWIW there is a draft proposal for an "after-build" hook: https://github.com/metalsmith/metalsmith/issues/392 which would allow easily running native CLI tools via metalsmith plugins (like exiftool, pngquant and jpegoptim) on the build output.

**Reply by rludgero** [reply]
> Ah thanks, but that looks like an app rather than a plugin CLI tool. And when I click through (Squish ->jSquash -> Squoosh), it seems the project is unmaintained. I'd been looking at replacing imagemin, but was looking more in the direction of sharp: https://github.com/lovell/sharp which also allows on-the-fly img transforms.
> 
> FWIW there is a draft proposal for an "after-build" hook: #392 which would allow easily running native CLI tools via metalsmith plugins (like exiftool, pngquant and jpegoptim) on the build output. 👍 

Sorry, my bad.  I believed it was possible to implement it without having a CLI.  :pray:

Yeah, sharp works great, I currently use the [metalsmith-sharp](https://github.com/axe312ger/metalsmith-sharp).

What happens is that I was impressed by the level of compression of Squish, as well as that of [onlineimagetool](https://www.onlineimagetool.com/en/) which is another web app that compresses images. That's why I thought of proposing it.
